### DISCUSSION TITLE
Google Takeout size vs. immich import size

### DISCUSSION THREAD

**Discussion by Psy185** [simulot/immich-go#161] [Q&A] [answered]
Hi and thanks for this great tool, you are making my life so much easier! :)

I was just wondering one thing: my Google Takeout had 220GB and after the import was completed, my immich library has 181GB (but it already had 30GB before, but most of that was also included in the takeout).

Where is this discrepancy coming from? Does Google create so much overhead/redundancy? It doesn't look like it to me.

Also, the immich jobs are still running for a while (thumbs, facial, storage template migration). 

Furthermore no albums were created from the takeout. But I think it should have happened, because -create-albums defaults to TRUE. Or should I've included -create-album-folder=TRUE?

This was the command for my import:
`immich-go -server http://localhost:2283/ -key **** upload -keep-untitled-albums takeout-*.zip`

**Comment by simulot** [comment, marked as answer]
> Hi and thanks for this great tool, you are making my life so much easier! :)

Thanks for you comment!

The discrepancy comes from the takeout process done by Google. I have observed that photos can be duplicated between albums and Year folders. More albums you have more redundancy you get. 

Yes, immich-server accepts photos at high pace and defers the heavy work to a queue. It could take hours to have process the backlog. 

Immich-go as immich-server detects duplicates and keep only one sample of the image. Immich-go uses this to create albums when the `-google-photos` flag is given. But, I don't see it in your command line.

The good new is that you can relaunch the same command + the google-photos flag. The albums will be created in immich. This process is very fast.
```
immich-go -server http://localhost:2283/ -key **** upload -google-photos -keep-untitled-albums takeout-*.zip
```

**Comment by Psy185** [comment]
Thanks for the quick reply :)

Damn, how could i forget the most important option flag lol... Anyway, good that it's possible afterwards. Should I wait for all Immich jobs to finish? It's still running and if it continues in this speed it will probably take a week or so. It's just running on an old Intel Nuc.

**Reply by simulot** [reply]
Immich server is doing lot of heavy work like face recognition, smart tagging based on IA model, reverse geotagging...

We have reports from users having 500K+ photos, running immich on under powered configuration. It takes time, but the work will be done.
